manifests/osfamily/solaris.pp

```diff
@@ -107,6 +107,12 @@
           logoutput   => 'on_failure',
           refreshonly => true,
         }
+        ~> exec { 'puppet_agent add publisher to pkg':
+          command     => "pkg set-publisher -e -g ${pkgrepo_dir} ${publisher}",
```

Comment:
If the `publisher` already exists (as described in https://tickets.puppetlabs.com/browse/PA-5215) will this command still work?

Reply:
ping @rstuart-indue

Reply:
Sorry @joshcooper  it looks like I missed the previous notifications. 

I don't seem to be able to find the linked ticket above. The pkg man page says it will update or add. here is a system and executing this command when the publisher already exists:
`
[16:32:27] root@host
/root# pkg publisher
PUBLISHER                   TYPE     STATUS P LOCATION
solaris                     origin   online F http://172.20.98.16:8003/IPS/
puppetlabs.com              origin   online F file:///etc/puppetlabs/installer/solaris.repo/
[16:32:33] root@host
/root# pkg publisher puppetlabs.com

            Publisher: puppetlabs.com
                Alias:
           Origin URI: file:///etc/puppetlabs/installer/solaris.repo/
        Origin Status: Online
              SSL Key: None
             SSL Cert: None
          Client UUID: 6cfd453a-2141-11ee-9502-43f3b6da7299
      Catalog Updated: Thu Nov  9 03:44:37 2023
    Publisher enabled: Yes
[16:32:36] root@host
/root# pkg set-publisher -e -g /etc/puppetlabs/installer/solaris.repo/ puppetlabs.com
[16:32:44] root@host
/root# pkg publisher puppetlabs.com
            Publisher: puppetlabs.com
                Alias:
           Origin URI: file:///etc/puppetlabs/installer/solaris.repo/
        Origin Status: Online
              SSL Key: None
             SSL Cert: None
          Client UUID: 6cfd453a-2141-11ee-9502-43f3b6da7299
      Catalog Updated: Thu Nov  9 03:44:37 2023
    Publisher enabled: Yes
[16:32:51] root@host
`

This additional code is to have Solaris 11 behave the same was as redhat and configure the repo as appropriate. ref https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/main/manifests/osfamily/redhat.pp#L162 and this block mimics the yumrepo declaration of line 167.